src/utils/sentry/httpclient.ts

```diff
@@ -331,41 +383,51 @@ function _wrapXHR(client: Client, options: HttpClientOptions): void {
     try {
       _xhrResponseHandler(options, xhr, method, headers)
     } catch (e) {
-      DEBUG_BUILD && logger.warn('Error while extracting response event form XHR response', e)
+      logger.warn('Error while extracting response event form XHR response', e)
     }
   })
 }
 
 /**
- * Checks whether to capture given response as an event
+ * Checks whether to capture given response as an event.
  *
- * @param status response status code
- * @param url response url
+ * @param options HttpClientOptions containing status codes, targets, and deny URLs.
+ * @param status Response status code.
+ * @param url Response URL.
+ * @returns true if the response should be captured, false otherwise.
  */
 function _shouldCaptureResponse(options: HttpClientOptions, status: number, url: string): boolean {
   return (
-    _isInGivenStatusRanges(options.failedRequestStatusCodes, status) &&
     _isInGivenRequestTargets(options.failedRequestTargets, url) &&
+    _isOutGivenDenyUrls(options.denyUrls, url) &&
+    _isInGivenStatusRanges(options.failedRequestStatusCodes, status) &&
     !isSentryRequestUrl(url, getClient())
   )
 }
-
 /**
  * Creates a synthetic Sentry event from given response data
  *
  * @param data response data
  * @returns event
  */
 function _createEvent(data: {
-  url: string
-  method: string
-  status: number
-  responseHeaders?: Record<string, string>
-  responseCookies?: Record<string, string>
-  requestHeaders?: Record<string, string>
-  requestCookies?: Record<string, string>
+  request: {
+    url: string
+    method: string
+    requestHeaders?: Record<string, string>
+    requestCookies?: Record<string, string>
+    data?: any
```

Comment:
Not an ommitted type, this is matching https://github.com/getsentry/sentry-javascript/blob/4dcfb2483b01e583e12de970d5dbeddc2159c240/packages/types/src/request.ts#L5